hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmLifecycleRuleAndOperator.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.helpers;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+
+/**
+ * A class that encapsulates lifecycleRule andOperator.
+ */
+public final class OmLifecycleRuleAndOperator {
+
+  private final Map<String, String> tags;
+  private final String prefix;
+
+  private OmLifecycleRuleAndOperator(Map<String, String> tags, String prefix) {
+    this.tags = tags;
+    this.prefix = prefix;
+  }
+
+  @Nonnull
+  public Map<String, String> getTags() {
+    return tags;
+  }
+
+  @Nullable
+  public String getPrefix() {
+    return prefix;
+  }
+
+  /**
+   * Validates the OmLifecycleRuleAndOperator.
+   * Ensures the following:
+   * - Either tags or prefix must be specified.
+   * - If there are tags and no prefix, the tags should be more than one.
+   * - Prefix alone is not allowed.
+   *
+   * @throws OMException if the validation fails.
+   */
+  public void valid() throws OMException {
+    if ((tags == null || tags.isEmpty()) && (prefix == null || prefix.isEmpty())) {
+      throw new OMException("Invalid lifecycle rule andOperator configuration: " +
+          "Either 'Tags' or 'Prefix' must be specified.",
+          OMException.ResultCodes.INVALID_REQUEST);
+    }
+
+    if (tags != null && !tags.isEmpty()) {
+      if (prefix == null || prefix.isEmpty()) {
+        if (tags.size() == 1) {
+          throw new OMException("Invalid lifecycle rule andOperator configuration: " +
+              "If 'Tags' are specified without 'Prefix', there should be more than one tag.",
+              OMException.ResultCodes.INVALID_REQUEST);
+        }
+      }
+    }
+
+    if (prefix != null && !prefix.isEmpty() && (tags == null || tags.isEmpty())) {
+      throw new OMException("Invalid lifecycle rule andOperator configuration: " +
+          "'Prefix' alone is not allowed.",
+          OMException.ResultCodes.INVALID_REQUEST);
+    }
+  }
+
+
+  /**
+   * The builder for the OmLifecycleRuleAndOperator class.
+   */
+  public static class Builder {
+    private Map<String, String> tags = new HashMap<>();
+    private String prefix;
+
+    public Builder setPrefix(String lcPrefix) {
+      this.prefix = lcPrefix;
+      return this;
+    }
+
+    public Builder addTag(String key, String value) {
+      this.tags.put(key, value);
+      return this;
+    }
+
+    public Builder setTags(Map<String, String> lcTags) {
+      if (lcTags != null) {
+        this.tags = new HashMap<>(lcTags);
+      }
+      return this;
+    }
+
+    public OmLifecycleRuleAndOperator build() {
+      return new OmLifecycleRuleAndOperator(this.tags, this.prefix);
+    }
+  }
+
+  @Override
+  public String toString() {
+    return "OmLifecycleRuleAndOperator{" +
+        "prefix='" + prefix + '\'' +
+        ", tags=" + tags +
```

Comment:
I'm afraid that directly use tags will not print all info as expected.

Reply:
The `HashMap` toString will print the key and value